spec/statement.dd

```diff
@@ -296,18 +294,28 @@ $(H3 $(LNAME2 condition-variables, Condition Variables))
         $(P
         When an $(I Identifier) form of *IfCondition* is used, a
         variable is declared with that name and initialized to the
-        value of the *Expression*.
+        value of the $(I Expression).
         )
 
-        * If $(D auto) $(I Identifier) is provided, the type of the variable
-          is the same as *Expression*.
+        *   If both `auto` and `ref` are present among $(I IfConditionStorageClasses),
+            the variable is a reference
+            if and only if the $(I Expression) is an lvalue;
```

Comment:
On second thought, given that the DIP doesn't describe this, and the implementation doesn't implement it, and Walter has described his dislike for `auto ref` in the past, I don't think the spec should define `auto ref` semantics for `if`, so I don't think this PR should elaborate on it.

I'll ask Walter what the intended behavior is here.

Reply:
He said pulled spec changes shouldn't predate an implementation, and it should be discussed on the news group first, so I suggest removing this specification.

Reply:
Implementation is coming (https://github.com/dlang/dmd/pull/16771)